### DISCUSSION TITLE
GChat notification channel

### DISCUSSION THREAD

**Discussion by abhayshankar** [ludeknovy/jtl-reporter#134] [Ideas]
Currently we have only MS Notification channel in External notifications which can be configured.
Can we include GChat Notification channel as well ?

**Comment by ludeknovy** [comment]
Hi @abhayshankar, yes that should be doable. Do you want to take a part in the implementation? Eg re-searching the webhook for google chat or any other way how to achieve it. How does it sound to you?

**Reply by abhayshankar** [reply]
Sure @ludeknovy , I will have a look into this.

**Comment by vijopv83** [comment]
We can create a Web-hook in Google Chat, then we have to pass the message as a Body in a POST Request.
Sharing a sample Body for Google Chat - Card Type:

`{
  "cards": [
    {
      "header": {
        "title": "JTL Reporter",
        "subtitle": "Performance Scenario: App Login",
        "imageUrl": "https://opengraph.githubassets.com/028ad554cd7be63e3b8905b64fd88426bd06bb6b931513335a3529fd7d35ef35/ludeknovy/jtl-reporter-be"
      },
      "sections": [
        {
          "widgets": [
              {
                "keyValue": {
                  "topLabel": "Test Status",
                  "content": "Passed"
                  }
              },
              {
                "keyValue": {
                  "topLabel": "Error Rate",
                  "content": "0 %"
                }
              },
              {
                "keyValue": {
                  "topLabel": "90% percentile",
                  "content": "48 ms"
                }
              },
              {
                "keyValue": {
                  "topLabel": "Throughput",
                  "content": "37.16 reqs/s"
                }
              },
              {
                "keyValue": {
                  "topLabel": "Duration",
                  "content": "34.97 min"
                }
              }
          ]
        },
        {
          "widgets": [
              {
                  "buttons": [
                    {
                      "textButton": {
                        "text": "OPEN RESULTS",
                        "onClick": {
                          "openLink": {
                            "url": "https://github.com/ludeknovy/jtl-reporter"
                          }
                        }
                      }
                    }
                  ]
              }
          ]
        }
      ]
    }
  ]
}`

**Reply by ludeknovy** [reply]
Hi @vijopv83 
Thanks for the payload. I will have a look.

**Comment by vijopv83** [comment]
<img width="335" alt="GChat" src="https://user-images.githubusercontent.com/65507674/181155278-27039e55-3845-4e21-91f9-ee47ea89ab29.png">

Sharing a sample of the above chat in GChat.
We can redirect to the specific report once it is imported.
